meetings/2021-07/july-14.md

```diff
@@ -0,0 +1,1028 @@
+# 14 July, 2021 Meeting Notes
+-----
+
+**Remote attendees:** 
+| Name                 | Abbreviation   | Organization       |
+| -------------------- | -------------- | ------------------ |
+| Waldemar Horwat      | WH             | Google             |
+| Istvan Sebestyen     | IS             | Ecma International |
+| Richard Gibson       | RGN            | OpenJS Foundation  |
+| Jack Works           | JWK            | Sujitech           |
+| Josh Blaney          | JPB            | Apple              |
+| Chengzhong Wu        | CZW            | Alibaba            |
+| Devin Rousso         | DRO            | Apple              |
+| Philip Chimento      | PFC            | Igalia             |
+| Jamie Kyle           | JK             | Rome               |
+
+
+## Ergonomic Brand Checks for Stage 4
+Presenter: Jordan Harband (JHD)
+
+- [proposal](https://github.com/tc39/proposal-private-fields-in-in/)
+- [slides](https://github.com/tc39/proposal-private-fields-in-in/issues/7)
+
+JHD: So, as a quick refresher, class fields is a stage 4 feature. However, if you would like to determine if an object has a private field, currently you have to use try-catch. This proposal gives you a straightforward boolean check of whether a private field is in object and that includes fields, private methods, and private accessors. The proposal stage 3, and I have all of these check marks here to measure progress toward stage four. So, as of this morning, Firefox version 90 is available Chrome. 91 has been available for a while. With this feature. The features also in webkit nightly and should be in the next Safari technology preview, which I believe will make it into the next full version of safari. But of course, Apple never discloses that before it's released. It's in node as a version 16.4.0. And it's also will be default. It's been in Babel for a while, it will be on by default with the end preset as soon as someone releases the package and merges the pr, which will happen after this feature gets stage the pull request of the spec exists and is approved. Test 262 test also have been landed. And so based on all of that I believe it clearly meets the criteria for stage four and want to ask the committee for consensus for that.
```

Comment:
```suggestion
JHD: So, as a quick refresher, class fields is a stage 4 feature. However, if you would like to determine if an object has a private field, currently you have to use try-catch. This proposal gives you a straightforward boolean check of whether a private field is in an object and that includes fields, private methods, and private accessors. The proposal is stage 3, and I have all of these checkmarks here to measure progress toward stage 4. So, as of this morning, Firefox version 90 is available, and Chrome 91 has been available for a while, with this feature. The feature is also in Webkit Nightly and should be in the next Safari Technology Preview, which I believe will make it into the next full version of Safari. (But of course, Apple never discloses that before it's released). It's in node as of v16.4.0. It's been in Babel for a while; it will be on by default with the env preset as soon as someone releases the package and merges the pr, which will happen after this feature gets stage 4. The pull request to the spec exists and is approved. test262 tests also have been landed. And so based on all of that I believe it clearly meets the criteria for stage 4 and want to ask the committee for consensus for that.
```